cal/api/views.py

```diff
@@ -201,14 +169,17 @@ class ColorCategoryDetailEvents(generics.ListAPIView):
     serializer_class = GEventSerializer
 
     def get_queryset(self):
-        return ColorCategory.objects.get(user=self.request.user, id=self.kwargs['pk']).query()
+        category = ColorCategory.objects.get(user=self.request.user, id=self.kwargs['pk'])
+        return category.query()
 
 
 class ColorCategoryDetailEventWeek(APIView):
 
+    serializer_class = ColorCategoryList
+
     def get(self, request, *args, **kwargs):
         category = ColorCategory.objects.get(user=self.request.user, id=self.kwargs['pk'])
-        return Response(category.get_hours_per_week())
+        return Response(category.get_hours_per_week(self.request.query_params.get('timezone')))
```

Comment:
Be wary of the future when we want to be able to get the hours per time period for events within a certain time range. For example, if we had a more detailed "day" view in our chart and scrolled back in time, and now need to hit our api with a start and end time. This would mean we have to specify the start and end time, which may or may not have a timezone attached to it. Given that the timezone itself may or may not be specified, how do we managed the relationship between the two?

`handle_time_string` already covers most of what I just said, I just wanted to give you a heads up.